internal/server/error_handling.go

```diff
@@ -0,0 +1,79 @@
+package server
+
+import (
+	"bytes"
```

Comment:
```suggestion
	"strings"
```

Reply:
For an initial implementation, `strings.Builder` and `bytes.Buffer` are _mostly_ the same for this use case. My main reasoning here is that this function could expand into more stuff in the future and I don't want to keep it locked down to be just a `strings.Builder`. There are a few subtleties about builders:

* You can't access the underlying bytes the same way you do with the buffer, which I might want to in the future
* Reset functions are quite different among them: buffer reset rewinds and reuses the underlying byte buffer, while builder reset sends it for GC so you'll need a new allocation in the future

Since the speed performance is somewhat negligible, I'm opting to keep it like this but I'll keep this in mind for future stuff 😁